yt_dlp/extractor/mave.py

```diff
@@ -1,24 +1,75 @@
-import re
+import functools
+import math
 
 from .common import InfoExtractor
 from ..utils import (
+    InAdvancePagedList,
     clean_html,
     int_or_none,
     parse_iso8601,
+    traverse_obj,
     urljoin,
 )
-from ..utils.traversal import require, traverse_obj
 
 
-class MaveIE(InfoExtractor):
-    _VALID_URL = r'https?://(?P<channel>[\w-]+)\.mave\.digital/(?P<id>ep-\d+)'
+class MaveBaseIE(InfoExtractor):
+    _API_BASE_URL = 'https://api.mave.digital/v1/website'
+    _API_BASE_STORAGE_URL = 'https://store.cloud.mts.ru/mave/'
+
+    def _load_channel_meta(self, channel_id):
+        return self._download_json(
+            f'{self._API_BASE_URL}/{channel_id}/', channel_id,
+            note='Downloading channel metadata')
+
+    def _load_episode_meta(self, channel_id, episode_code):
+        return self._download_json(
+            f'{self._API_BASE_URL}/{channel_id}/episodes/{episode_code}',
+            episode_code, note='Downloading episode metadata')
+
+    def _create_entry(self, channel_id, channel_meta, episode_meta):
+        return {
+            'display_id': f'{channel_id}-{episode_meta["code"]}',
+            'channel_id': channel_id,
+            'channel_url': f'https://{channel_id}.mave.digital/',
+            'vcodec': 'none',
+            **traverse_obj(episode_meta, {
+                'id': ('id', {str}),
+                'url': ('audio', {urljoin(self._API_BASE_STORAGE_URL)}),
+                'title': ('title', {str}),
+                'description': ('description', {clean_html}),
+                'thumbnail': ('image', {urljoin(self._API_BASE_STORAGE_URL)}),
+                'duration': ('duration', {int_or_none}),
+                'season_number': ('season', {int_or_none}),
+                'episode_number': ('number', {int_or_none}),
+                'view_count': ('listenings', {int_or_none}),
+                'like_count': ('reactions',
+                               lambda _, v: v['type'] == 'like',
+                               'count', {int_or_none}, any),
+                'dislike_count': ('reactions',
+                                  lambda _, v: v['type'] == 'dislike',
+                                  'count', {int_or_none}, any),
```

Comment:
Please keep the formatting as I suggested it
```suggestion
                'like_count': ('reactions', lambda _, v: v['type'] == 'like', 'count', {int_or_none}, any),
                'dislike_count': ('reactions', lambda _, v: v['type'] == 'dislike', 'count', {int_or_none}, any),
```